lib/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -254,43 +254,49 @@ public void run() {
 	 * transport events/notifications
 	 ***************************************/
 
-	void onMessage(ProtocolMessage message) {
-		if(protocolListener != null)
-			protocolListener.onRawMessage(message);
-		switch(message.action) {
-		case HEARTBEAT:
-			onHeartbeat(message);
-			break;
-		case ERROR:
-			ErrorInfo reason = message.error;
-			if(reason == null)
-				Log.e(TAG, "onMessage(): ERROR message received (no error detail)");
-			else
-				Log.e(TAG, "onMessage(): ERROR message received; message = " + reason.message + "; code = " + reason.code);
+	void onMessage(ProtocolMessage message) throws AblyException {
```

Comment:
Over to you @paddybyers as I don't have a strong opinion on this especially because it's Java